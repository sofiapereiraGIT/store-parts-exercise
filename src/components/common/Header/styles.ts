import styled from "styled-components";

export const Header = {
  Header: styled.header`
    background-color: black;
    color: white;
    display: flex;
  `,
  Title: styled.h1`
    margin-left: 0.67em;
    cursor: pointer;
  `,
};
