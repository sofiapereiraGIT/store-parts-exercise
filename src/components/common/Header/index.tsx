import { useHistory } from "react-router-dom";
import { Header as S } from "./styles";

export default function Header() {
  const history = useHistory();

  return (
    <S.Header>
      <S.Title data-testid="headerTitle" onClick={() => history.push("/")}>
        Store Parts
      </S.Title>
    </S.Header>
  );
}
