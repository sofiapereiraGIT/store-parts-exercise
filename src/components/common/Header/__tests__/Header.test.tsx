import React from "react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react";
import Header from "../index";

describe("Header Tests", () => {
  it("Renders properly", () => {
    const renderedComponent = renderer.create(<Header />);
    expect(renderedComponent.toJSON()).toMatchSnapshot();
  });

  it("Redirects properly to '/' when title is clicked", () => {
    const history = createMemoryHistory();
    const pushSpy = jest.spyOn(history, "push");
    const { getByTestId } = render(
      <Router history={history}>
        <Header />
      </Router>
    );

    const headerTitle = getByTestId("headerTitle");
    fireEvent.click(headerTitle);
    expect(pushSpy).toHaveBeenCalledWith(`/`);
  });
});
