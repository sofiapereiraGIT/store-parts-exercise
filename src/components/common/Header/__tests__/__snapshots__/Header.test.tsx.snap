// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Tests Renders properly 1`] = `
<header
  className="sc-bczRLJ cMfLWl"
>
  <h1
    className="sc-gsnTZi bVxfwA"
    data-testid="headerTitle"
    onClick={[Function]}
  >
    Store Parts
  </h1>
</header>
`;
